common/nodejs-utils/src/bin/update_node_inventory.rs

```diff
@@ -47,11 +41,11 @@ fn main() -> Result<()> {
 
 fn write_inventory(
     inventory_path: impl Into<PathBuf>,
-    upstream_artifacts: &HashSet<Artifact<Version, Sha256>>,
+    upstream_artifacts: &[Artifact<Version, Sha256, Option<()>>],
```

Comment:
Why doesn't `Artifact` implement `Hash`?

Reply:
It was [removed a while back](https://github.com/Malax/inventory/commit/502319a1a166216d9d71b9bb58ccc8749a17f5d4). I implemented in the go inventory, but it was admittedly a bit of a hack, and not particularly intuitive that an `Artifact` would automatically implement `Hash` especially based on the checksum value alone.